py/mphal.h

```diff
@@ -66,4 +66,11 @@ mp_uint_t mp_hal_ticks_ms(void);
 mp_uint_t mp_hal_ticks_us(void);
 #endif
 
+#ifndef mp_hal_pin_obj_t
+#define mp_hal_pin_obj_t mp_obj_t
+#define mp_hal_get_pin_obj(pin) (pin)
```

Comment:
> which validate pin once on entry and then use it many times w/o checking overhead.

This would be achieved by checking only in the mp_hal_get_pin_obj function (which is executed once at the start, like mp_obj_get_int) and then all accessor functions like read/write don't need to check.